packages/orm/src/drivers/PostgresDriver.ts

```diff
@@ -379,31 +387,50 @@ async function batchUpdate(knex: Knex, meta: EntityMetadata<any>, entities: Enti
                   const maybeArray = c.serde instanceof EnumArrayFieldSerde ? "[]" : "";
                   return `?::${c.dbType}${maybeArray}`;
                 })
-                .join(",")})`,
+                .join(", ")}, ?::timestamp)`,
           )
           .join(",")}
-      ) AS data(${columns.map((c) => c.columnName).join(",")})
-      WHERE ${meta.tableName}.id = data.id
+      ) AS data(${columns.map((c) => c.columnName).join(",")},original_updated_at)
+      WHERE
+        ${meta.tableName}.id = data.id
+        AND date_trunc('milliseconds', ${meta.tableName}.updated_at) = data.original_updated_at
+      RETURNING ${meta.tableName}.id
    `;
   } else {
     // Issue 1 UPDATE statement with bindings having an array for each column's new values
     bindings = columns.map(() => []);
+    // Add 1 more column for original_updated_at
+    const originalUpdatedAt: Date[] = [];
+    bindings.push(originalUpdatedAt);
```

Comment:
Our `UPDATE`s work by basically uploading a mini-table of the changed rows, and this added `originalUpdatedAt` as a new "column" in that mini-table (`data`), which we don't use in the `SET ...` statements, but we do use in the `WHERE` clause (just like the `id` column).